src/v/cluster/scheduling/leader_balancer.cc

```diff
@@ -276,8 +277,20 @@ void leader_balancer::trigger_balance() {
     });
 }
 
-ss::future<ss::stop_iteration> leader_balancer::balance() {
+bool leader_balancer::should_stop_balance() const {
     if (!_enabled()) {
+        return true;
+    }
+
+    if (_as.local().abort_requested()) {
+        return true;
+    }
+
+    return false;
```

Comment:
nit: return !_enabled || _as.local().abort_requested();

Reply:
Switched to recommendation, good call